contracts/Registry.vy

```diff
@@ -148,17 +169,21 @@ def _get_rates(_pool: address) -> uint256[MAX_COINS]:
     rates: uint256[MAX_COINS] = empty(uint256[MAX_COINS])
     base_pool: address = self.pool_data[_pool].base_pool
     if base_pool == ZERO_ADDRESS:
-        rate_method_id: Bytes[4] = slice(self.pool_data[_pool].rate_method_id, 28, 4)
+        rate_info: bytes32 = self.pool_data[_pool].rate_info
+        rate_calc_addr: uint256 = convert(slice(rate_info, 8, 20), uint256)
+        rate_method_id: Bytes[4] = slice(rate_info, 28, 4)
 
         for i in range(MAX_COINS):
             coin: address = self.pool_data[_pool].coins[i]
             if coin == ZERO_ADDRESS:
                 break
-            if coin == self.pool_data[_pool].ul_coins[i]:
+            if rate_info == EMPTY_BYTES32 or coin == self.pool_data[_pool].ul_coins[i]:
```

Comment:
This should cover all the grounds:
1. rate_info is EMPTY_BYTES32
2. rate_info only has a rate_calculator address
3. rate_info only has a rate_method_id -> this gets caught by the else clause
4. rate_info has both -> this will get caught by 2